src/commons/utils/corio_utils.py

```diff
@@ -265,6 +275,52 @@ def setup_environment():
     os.makedirs(DATA_DIR_PATH, exist_ok=True)
 
 
+def activate_degraded_mode(options: dict) -> None:
+    """Create bucket and degrade node as needed .
+
+    :param options : dictionary to fetch command line params
+    :return : None
+    """
+    bucket = f'object-op-{time.perf_counter_ns()}'.lower()
+    response = S3Bucket(access_key=options.access_key,
+                        secret_key=options.secret_key,
+                        endpoint_url=options.endpoint_url,
+                        use_ssl=options.use_ssl).create_bucket(bucket)
+    if response:
+        os.environ["d_bucket"] = bucket
+
+    if options.degrade_pod:
+        __degrade_nodes(options)
+
+
+def __degrade_nodes(options: dict) -> None:
+    """
+    degrade nodes as needed for execution by shutdown/deleting pod
+    :param options : dictionary to fetch command line params
+    """
+    nodes = CLUSTER_CFG["nodes"]
+    (host, user, password) = (None, None, None)
+    for node in nodes:
+        if node["node_type"] == "master":
+            host = node["hostname"]
+            user = node["username"]
+            password = node["password"]
+            break
+    logical_node = LogicalNode(host=host,
+                               user=user,
+                               password=password)
+    LOGGER.info("Shutdown the data pods safely by making replicas=0")
+    pod_list = logical_node.get_all_pods(pod_prefix=DATA_POD_NAME_PREFIX)
```

Comment:
can we take pod_prefix as arg as to shutdown data pod or server pod?

Reply:
Yes, Changed accordingly for data and server pods, kept data pod as default